### ISSUE TITLE
Duplicate container instances

### ISSUE THREAD

**Issue by gmponos** [liip/LiipFunctionalTestBundle#350]
Hello there,

According to [symfony official documentation](https://symfony.com/doc/current/testing.html#your-first-functional-test) in order to test a controller you have to initialize a client like this:

```
        $client = static::createClient();
```

And this is what I've been doing.

after creating my client I used to do something like this:

```
        $this->loadFixtures($this->getFixtureData());
```

Then today I reached my memory limit and dived little deeper into this bundle. I realized that the reason was that there is a container loaded in ` $client = static::createClient();` and also `loadFixtures` is loading a second container.

After making some copy-paste custom code I changed the signature of `loadFixtures` to accept a container and the memory for PHP 5 dropped from ~500 MB to ~200 MB.

In my belief it would've been better if you let the developer set his own container somehow.

You could say that I could use the `makeClient` instead but unfortunately make client is doing the same thing and it loading a second container too by calling internally this 

```
$client = static::createClient(array('environment' => $this->environment), $params)
```

**Comment by Jean85** [comment]
The fact that the client is loading a second container is absolutely correct. You don't want to use the same one, since the first one is spun up by the test and it's already booted, and it can be tainted by any modification or operation that you're doing to prepare your test case.

**Comment by gmponos** [comment]
The work around that I found was this:

```
    public function setUp()
    {
        parent::setUp();
        $this->client = static::createClient();
        $this->containers['|test'] = $this->client->getContainer();
        $this->loadFixtures($this->getFixtureData());
    }
```

I feel it's kinda dirty and developers can easily forget to do this. Unfortunately it seems that I will need to either fork the project or go with a custom solution because it's way costly.

@Jean85 do you suggest that the client should not load a container or that LiipBundle should not load a second one?

**Comment by Jean85** [comment]
Nope, I suggest neither. You shouldn't mix those two different container, they are intended to stay completely separate! You're trying to optimize something that you shouldn't. And believe me, I'm speaking from experience, I've tried to do the same in the past, obtaining only issues and unreliable tests.

The system under test need it's own container, and it's inside the client.
You (may) need a container inside the tests, to manipulate the state of your app before the test, or to make assertions after, but you have to use a separate one.

**Comment by gmponos** [comment]
> You (may) need a container inside the tests, to manipulate the state of your app before the test,

Are you talking here about the one that is inside the client? if yes... isn't this what the `loadFixtures` is doing? Manipulating the state of my app?
